src/test/scala/com/cognite/sdk/scala/common/StringDataPointsResourceBehaviors.scala

```diff
@@ -27,42 +27,41 @@ trait StringDataPointsResourceBehaviors extends Matchers with RetryWhile { this:
 
         retryWithExpectedResult[Seq[StringDataPointsByIdResponse]](
           dataPoints.queryStringsById(stringTimeSeriesId, start, end.plusMillis(1)),
-          None,
-          Seq(p => p.head.datapoints should have size testStringDataPoints.size.toLong)
+          p => p.head.datapoints should have size testStringDataPoints.size.toLong
         )
 
         retryWithExpectedResult[Option[StringDataPoint]](
           dataPoints.getLatestStringDataPointById(stringTimeSeriesId),
-          None,
-          Seq(dp => dp.isDefined shouldBe true, dp => testStringDataPoints.toList should contain(dp.get))
+          { dp =>
+            dp.isDefined shouldBe true
```

Comment:
Are you sure this doesn't just define something, but doesn't execute it?
Maybe try changing this to "shouldBe false" and verify that it does get tested, and that it doesn't just "assert" the return value from this function (ie. the next line).

Reply:
Yep, I'm quite sure about it.

1. The `should` things just throw an exception when it does not match and AFAIK Scala is eagerly evaluated (otherwise all tests would be broken). When it's OK, it returns the `Assertion` (they probably meant this as a typed Unit?)
2. I have tried prepending `true shouldBe false` and then the test fail.

Reply:
Ok, that's cool. The usual way to write this would be
```scala
dp => {
  dp.isDefined shouldBe true
  testStringDataPoints.toList should contain(dp.get)
}
```

If that works and gives proper error messages I'm fine with removing the `Seq`, and I also prefer it. :)

Reply:
> The usual way to write this would be

Ok, fine. I found out both work, but they used the `{ x => ... }` for block lambdas in the docs, so I though this is the Scala way :) The `x => { ... }` however also feels more natural to me, so I'll switch that.